src/transformers/tokenization_marian.py

```diff
@@ -0,0 +1,162 @@
+    model_input_names = ["attention_mask"]  # actually attention_mask, decoder_attention_mask
+
+    def __init__(
+        self,
+        vocab=None,
+        source_spm=None,
+        target_spm=None,
+        source_lang=None,
+        target_lang=None,
+        unk_token="<unk>",
+        eos_token="</s>",
+        pad_token="<pad>",
+        max_len=512,
+    ):
+
+        super().__init__(
+            # bos_token=bos_token,
+            max_len=max_len,
+            eos_token=eos_token,
+            unk_token=unk_token,
+            # sep_token=sep_token,
+            # cls_token=cls_token,
+            pad_token=pad_token,
+            # mask_token=mask_token,
+            # **kwargs,
+        )
+        self.encoder = load_json(vocab)  # {k: int(v) for k, v in load_json(vocab).items()}
+        assert self.pad_token in self.encoder
+        self.decoder = {v: k for k, v in self.encoder.items()}
+
+        self.source_lang = source_lang
+        self.target_lang = target_lang
+
+        # load SentencePiece model for pre-processing
+        self.paths = {}
+
+        self.spm_source = sentencepiece.SentencePieceProcessor()
+        self.spm_source.Load(source_spm)
+
+        self.spm_target = sentencepiece.SentencePieceProcessor()
+        self.spm_target.Load(target_spm)
+
+        # Note(SS): splitter would require lots of book-keeping.
+        # self.sentence_splitter = MosesSentenceSplitter(source_lang)
+        try:
+            from mosestokenizer import MosesPunctuationNormalizer
+
+            self.punc_normalizer = MosesPunctuationNormalizer(source_lang)
+        except ImportError:
+            warnings.warn("Recommended: pip install mosestokenizer")
+            self.punc_normalizer = lambda x: x
+
+    def _convert_token_to_id(self, token):
+        return self.encoder[token]
+
+    def _tokenize(self, text: str, src=True) -> List[str]:
+        spm = self.spm_source if src else self.spm_target
+        return spm.EncodeAsPieces(text)
+
+    def _convert_id_to_token(self, index: int) -> str:
+        """Converts an index (integer) in a token (str) using the encoder."""
+        return self.decoder.get(index, self.unk_token)
+
+    def convert_tokens_to_string(self, tokens: List[str]) -> str:
+        """Uses target language sentencepiece model"""
+        return self.spm_target.DecodePieces(tokens)
+
+    def _append_special_tokens_and_truncate(self, tokens: str, max_length: int,) -> List[int]:
+        ids: list = self.convert_tokens_to_ids(tokens)[:max_length]
+        return ids + [self.eos_token_id]
+
+    def build_inputs_with_special_tokens(self, token_ids_0, token_ids_1=None) -> List[int]:
+        """
+        Build model inputs from a sequence or a pair of sequence for sequence classification tasks
+        by concatenating and adding special tokens.
+        A RoBERTa sequence has the following format:
+            single sequence: <s> X </s>
+            pair of sequences: <s> A </s></s> B </s>
```

Comment:
Doc seems wrong here (RoBERTa -> Marian?, sentence building doesn't seem to reflect the logic)

Reply:
good catch!